parser/src/parsercontext.cpp

```diff
@@ -0,0 +1,81 @@
+#include <boost/filesystem.hpp>
+
+#include <model/file.h>
+#include <model/file-odb.hxx>
+
+#include <util/hash.h>
+#include <util/odbtransaction.h>
+
+#include <parser/parsercontext.h>
+#include <parser/sourcemanager.h>
+
+namespace po = boost::program_options;
+
+namespace cc
+{
+namespace parser
+{
+
+ParserContext::ParserContext(
+    std::shared_ptr<odb::database> db_,
+    SourceManager& srcMgr_,
+    std::string& compassRoot_,
+    po::variables_map& options_):
+      db(db_),
+      srcMgr(srcMgr_),
+      options(options_),
+      compassRoot(compassRoot_)
+{
+  std::unordered_map<std::string, std::string> fileHashes;
+
+  (util::OdbTransaction(this->db))([&]
+   {
+     // Fetch all files from SourceManager
+     std::vector<model::FilePtr> files = this->srcMgr.getAllFiles();
+     files.erase(std::remove_if(files.begin(), files.end(),
+                                [](const auto &item)
+                                {
+                                  return item->type == model::File::DIRECTORY_TYPE ||
+                                         item->type == model::File::BINARY_TYPE;
+                                }), files.end());
+
+     for (model::FilePtr file : files)
+     {
+       if (boost::filesystem::exists(file->path))
+       {
+         if (!this->fileStatus.count(file->path))
+         {
+           auto content = file->content.load();
```

Comment:
I take it that this *content* is the actual source code stored in the database. Do we really have to load this into memory just to compare if the local disk contains the same hash? The foreign key IIRC maps file content hashes to the CLOB. Perhaps `file->content.object_id()` works here, as the hash is the primary key on the content table.

Reply:
Good point :thumbsup:

Reply:
Fixed in 52b7b72b31bdc585c43440a29071f0842e4570b6.